pyvrp/cpp/search/SwapStar.cpp

```diff
@@ -206,32 +240,12 @@ Cost SwapStar::evaluate(Route *routeU,
         {
             Cost deltaCost = 0;
 
-            auto const uDemand = data.location(U->client()).demand;
-            auto const vDemand = data.location(V->client()).demand;
-            auto const loadDiff = uDemand - vDemand;
-
-            deltaCost += costEvaluator.loadPenalty(routeU->load() - loadDiff,
-                                                   routeU->capacity());
-            deltaCost -= costEvaluator.loadPenalty(routeU->load(),
-                                                   routeU->capacity());
-
-            deltaCost += costEvaluator.loadPenalty(routeV->load() + loadDiff,
-                                                   routeV->capacity());
-            deltaCost -= costEvaluator.loadPenalty(routeV->load(),
-                                                   routeV->capacity());
-
             deltaCost += removalCosts(routeU->idx(), U->client());
             deltaCost += removalCosts(routeV->idx(), V->client());
 
-            if (deltaCost >= 0)  // an early filter on many moves, before doing
-                continue;        // costly work determining insertion points
-
             auto [extraV, UAfter] = getBestInsertPoint(U, V, costEvaluator);
             deltaCost += extraV;
 
-            if (deltaCost >= 0)  // continuing here avoids evaluating another
-                continue;        // potentially costly insertion point below
```

Comment:
These checks are of dubious value. In runs on my laptop they seem to cut of a lot of pretty good solutions prematurely.

Reply:
Well, that seems like a premature conclusion. I'll revert this.